uqcsbot/starboard.py

```diff
@@ -1,244 +1,598 @@
 import os
-from typing import Optional
 from threading import Timer
+from typing import Tuple, List
+from zoneinfo import ZoneInfo
 
 import discord
+from discord import app_commands
 from discord.ext import commands
 
 from uqcsbot import models
-# needs to be models and not just starboard because of namespacing with this class
+
+
+class BlacklistedMessageError(Exception):
+    # always caught. used to differentiate "starboard message doesn't exist" and "starboard message is blacklisted"
+    pass
+
+
+class SomethingsFucked(Exception):
+    # never caught. used for bad db states, which should never occur, but just in case, y'know?
+    def __init__(
+        self,
+        client: discord.Client,
+        modlog: discord.TextChannel,
+        message: str,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(message, *args, **kwargs)
+        client.loop.create_task(modlog.send(f"Bad Starboard state: {message}"))
+
 
 class Starboard(commands.Cog):
-    CHANNEL_NAME = "starboard"
+    SB_CHANNEL_NAME = "starboard"
     EMOJI_NAME = "starhaj"
+    MODLOG_CHANNEL_NAME = "admin-alerts"
+    BRISBANE_TZ = ZoneInfo("Australia/Brisbane")
 
     def __init__(self, bot: commands.Bot):
         self.bot = bot
         self.base_threshold = int(os.environ.get("SB_BASE_THRESHOLD"))
         self.big_threshold = int(os.environ.get("SB_BIG_THRESHOLD"))
```

Comment:
Maybe change this to `pin_threshold`

Reply:
My intention when writing was that the "big threshold" may not always mean pinning - in other servers, starboards have tiers where different emojis are edited into the starboard message, and committee may want to manually edit the pins or whatever. 

I can change big -> pin globally if it's a significant benefit to the readability, but the thresholds have deliberately action-agnostic names.

Reply:
Now that you've stated it, I agree with the action agnostic names. With the documentation you linked above (which I didn't see when I reviewed), this looks fine as is.